# HG changeset patch
# Parent  2ab2e5e7b0c0bb0b9614b4e03090f3acea931bed
Remove the list_tags command

You can build the list of tags using get_light_state("*").

diff --git a/core/jsonrpc.c b/core/jsonrpc.c
--- a/core/jsonrpc.c
+++ b/core/jsonrpc.c
@@ -973,8 +973,7 @@
         LGTD_JSONRPC_METHOD(
             "get_light_state", 1, // t
             lgtd_jsonrpc_check_and_call_get_light_state
-        ),
-        LGTD_JSONRPC_METHOD("list_tags", 0, lgtd_proto_list_tags),
+        )
     };
 
     assert(client);
diff --git a/core/proto.c b/core/proto.c
--- a/core/proto.c
+++ b/core/proto.c
@@ -151,23 +151,6 @@
 }
 
 void
-lgtd_proto_list_tags(struct lgtd_client *client)
-{
-    lgtd_client_start_send_response(client);
-
-    LGTD_CLIENT_WRITE_STRING(client, "[");
-    struct lgtd_lifx_tag *tag;
-    LIST_FOREACH(tag, &lgtd_lifx_tags, link) {
-        LGTD_CLIENT_WRITE_STRING(client, "\"");
-        LGTD_CLIENT_WRITE_STRING(client, tag->label);
-        LGTD_CLIENT_WRITE_STRING(client, LIST_NEXT(tag, link) ? "\"," : "\"");
-    }
-    LGTD_CLIENT_WRITE_STRING(client, "]");
-
-    lgtd_client_end_send_response(client);
-}
-
-void
 lgtd_proto_get_light_state(struct lgtd_client *client,
                            const struct lgtd_proto_target_list *targets)
 {
diff --git a/core/proto.h b/core/proto.h
--- a/core/proto.h
+++ b/core/proto.h
@@ -39,4 +39,3 @@
 void lgtd_proto_power_on(struct lgtd_client *, const struct lgtd_proto_target_list *);
 void lgtd_proto_power_off(struct lgtd_client *, const struct lgtd_proto_target_list *);
 void lgtd_proto_get_light_state(struct lgtd_client *, const struct lgtd_proto_target_list *);
-void lgtd_proto_list_tags(struct lgtd_client *);
diff --git a/tests/core/proto/test_proto_list_tags.c b/tests/core/proto/test_proto_list_tags.c
deleted file mode 100644
--- a/tests/core/proto/test_proto_list_tags.c
+++ /dev/null
@@ -1,41 +0,0 @@
-#include "proto.c"
-
-#include "mock_client_buf.h"
-#include "tests_utils.h"
-#include "tests_proto_utils.h"
-
-int
-main(void)
-{
-    struct lgtd_client client;
-
-    lgtd_proto_list_tags(&client);
-    if (strcmp(client_write_buf, "[]")) {
-        errx(
-            1, "expected empty list of tags but got %s instead",
-            client_write_buf
-        );
-    }
-
-    reset_client_write_buf();
-    lgtd_tests_insert_mock_tag("test");
-    lgtd_proto_list_tags(&client);
-    if (strcmp(client_write_buf, "[\"test\"]")) {
-        errx(
-            1, "expected [\"test\"] but got %s instead",
-            client_write_buf
-        );
-    }
-
-    reset_client_write_buf();
-    lgtd_tests_insert_mock_tag("@_@");
-    lgtd_proto_list_tags(&client);
-    if (strcmp(client_write_buf, "[\"@_@\",\"test\"]")) {
-        errx(
-            1, "expected [\"@_@\",\"test\"] but got %s instead",
-            client_write_buf
-        );
-    }
-
-    return 0;
-}
